src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -243,6 +270,51 @@ void WatchHierarchy(IVsHierarchy hierarchyToWatch)
                 return projectIds.First();
             }
 
+            private void WatchAndSubscribeToHierarchies(string moniker, IVsHierarchy? hierarchy)
```

Comment:
this is the same logic as the method above, except that it only deals with the watching of hierarchies.